src/Kestrel.Transport.Abstractions/Internal/TransportConnection.cs

```diff
@@ -63,8 +55,14 @@ public TransportConnection()
 
         public CancellationToken ConnectionClosed { get; set; }
 
-        public virtual void Abort()
+        // DO NOT remove this override to ConnectionContext.Abort(). Doing so would cause
+        // any TransportConnection that does not override Abort() or calls base.Abort()
+        // to stack overflow when IConnectionLifetimeFeature.Abort() is called.
+        public override void Abort(ConnectionAbortedException abortReason)
         {
+            AbortImpl(abortReason);
         }
+
+        public abstract void AbortImpl(ConnectionAbortedException abortReason);
```

Comment:
@davidfowl I know this name is bad. Any suggestions? At least it's a pubternal type.

Reply:
Make it protected and rename it to `AbortCore`